src/services/codefixes/convertPromisesToAwaitAndAsync.ts

```diff
@@ -0,0 +1,106 @@
+namespace ts.codefix {
+    const fixId = "convertPromisesToAwaitAndAsync";
+    const errorCodes = [Diagnostics.This_may_be_converted_to_use_async_and_await.code];
+    registerCodeFix({
+        errorCodes,
+        getCodeActions(context: CodeFixContext) {
+            const changes = textChanges.ChangeTracker.with(context, (t) => convertToAsyncAwait(t, context.sourceFile, context.span.start, context.program.getTypeChecker()));
+            return [createCodeFixAction(fixId, changes, Diagnostics.Convert_to_use_async_and_await, fixId, Diagnostics.Convert_all_to_use_async_and_await)];
+        },
+        fixIds: [fixId],
+        getAllCodeActions: context => codeFixAll(context, errorCodes, (changes, err) => convertToAsyncAwait(changes, err.file!, err.start, context.program.getTypeChecker())),
+    });
+    function convertToAsyncAwait(changes: textChanges.ChangeTracker, sourceFile: SourceFile, position: number, checker: TypeChecker): void {
+        //get the function declaration - returns a promise
+        const funcToConvert = checker.getSymbolAtLocation(getTokenAtPosition(sourceFile, position, false)).valueDeclaration;
+        //add the async keyword
+        changes.insertModifierBefore(sourceFile, SyntaxKind.AsyncKeyword, funcToConvert)
+        
+       for(let child of funcToConvert.getChildren()){
+            if(child.kind === SyntaxKind.Block){
+                for(let stmt of (<Block>child).statements){
+                    let promiseCall:CallExpression = getPromiseCall(stmt);
+                    let newNode = parseCallback(promiseCall, checker);
+
+                    if(newNode){
+                       changes.replaceNodeWithNodes(sourceFile, stmt, newNode);
+                    }
+                }
+            }
+        }
+    }
+
+    function getPromiseCall(node:Node): CallExpression{
+        if(node.kind == SyntaxKind.CallExpression){
+            return node as CallExpression;
+        }
+
+        for( let child of node.getChildren().filter(node => isNode(node)) ){

```

Comment:
I think you're looking for the `forEachChild` in `parser.ts`. But if not you can also just use

```ts
return node.getChildren().find(isNode);
```

Reply:
`getChildren()` is the wrong method to use since it returns the tokens, such as an opening parenthesis.